apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor.tsx

```diff
@@ -16,8 +16,8 @@ export interface EntityEditorProps extends DraftLinkState {
   isDirty: boolean;
   entitySubgraph: Subgraph<EntityRootType>;
   setEntity: (entity: Entity) => void;
-  replaceWithLatestDbVersion: () => Promise<void>;
   readonly: boolean;
+  onEntityUpdated: (entity: Entity) => void;
```

Comment:
I've made `null` an option instead so that we don't have the pointless functions, in https://github.com/hashintel/hash/pull/4221/commits/ea027b0f80104e6c0c006aa2600acece428f14a8

I'd rather force passing the prop for now so that we explicitly consider what we want it to be when using the editor in new places (rather than potentially overlooking it).